stable/admin/templates/ingress.yaml

```diff
@@ -0,0 +1,83 @@
+{{- if .Values.admin.ingress }}
+{{- if eq (include "defaultfalse" .Values.admin.ingress.enabled) "true" }}
+{{- if .Values.admin.ingress.api }}
+{{- if .Values.admin.ingress.api.hostname }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ template "admin.fullname" . }}-api
+  annotations:
+    {{- toYaml .Values.admin.ingress.api.annotations | trim | nindent 4 }}
+  labels:
+    app: {{ template "admin.fullname" . }}
+    chart: {{ template "admin.chart" . }}
+    release: {{ .Release.Name | quote }}
+    domain: {{ .Values.admin.domain }}
+    group: nuodb
+spec:
+  {{- if .Values.admin.ingress.api.tls }}
+  tls:
+    - hosts:
+      - {{ .Values.admin.ingress.api.hostname | quote }}
+    {{- if .Values.admin.ingress.api.secretName }}
+      secretName: {{ .Values.admin.ingress.api.secretName }}
+    {{- end -}}
+  {{- end }}
+  {{- if .Values.admin.ingress.api.className }}
+  ingressClassName: {{ .Values.admin.ingress.api.className | quote }}
+  {{- end }}
+  rules:
+  - host: {{ .Values.admin.ingress.api.hostname | quote }}
+    http:
+      paths:
+      - path: {{ default "/" .Values.admin.ingress.api.path }}
```

Comment:
Would any path other than `/` work with `ssl-passthrough=true` for the haproxy-ingress? Also, what would be the use case here, `nuocmd` clients connecting from outside the Kubernetes cluster?

Reply:
> Would any path other than / work with ssl-passthrough=true for the haproxy-ingress?

Correct. The `path` doesn't have any effect when `ssl-passthrough=true` because the TLS can't be decrypted by the controller. However, for the REST service ssl-passthrough is not mandatory as it is just HTTP service so someone might want to disable in the NuoAdmin, use the Ingress controller to terminate TLS with the client, and use basic authentication. I decided to not assume the configuration but make it as flexible as possible.

> Also, what would be the use case here, nuocmd clients connecting from outside the Kubernetes cluster?

Correct. This exposes the Admin REST service.

Reply:
So does it make sense to have a path other than `/`? If `nuocmd` is being issued by an external client, then it would have to have the path prefix embedded in the `--api-server` value or something, e.g. `https://<ingress host>/<path>`. I doubt that this would actually work, since I'm sure that there are places where we assume that `--api-server` has the form `<protocol>://<host>:<port>`, `<host>:<port>`, or just `<host>`.

Reply:
That's a good point. I only tested with the default path (`/`), Admin without TLS, and TLS termination on the Ingress controller.
If we have some assumptions for the URL in the client, then it would be better to not expose the path.